internal/cmd/skupper/listener/listener_create.go

```diff
@@ -1,47 +1,197 @@
-			skupperCmd.InputToOptions()
 			utils.HandleError(skupperCmd.Run())
 		},
+		PostRunE: func(cmd *cobra.Command, args []string) error {
+			return skupperCmd.WaitUntilReady()
+		},
 	}
 
 	skupperCmd.CobraCmd = cmd
+	skupperCmd.AddFlags()
 
 	return &skupperCmd
 }
 
-func (cmd *CmdListenerCreate) NewClient(cobraCommand *cobra.Command, args []string) {}
-func (cmd *CmdListenerCreate) AddFlags()                                            {}
-func (cmd *CmdListenerCreate) ValidateInput(args []string) []error                  { return nil }
-func (cmd *CmdListenerCreate) InputToOptions()                                      {}
-func (cmd *CmdListenerCreate) Run() error                                           { fmt.Println("Not implemented yet."); return nil }
-func (cmd *CmdListenerCreate) WaitUntilReady() error                                { return nil }
+func (cmd *CmdListenerCreate) NewClient(cobraCommand *cobra.Command, args []string) {
+	kubeconfig := os.Getenv("KUBECONFIG")
+	cli, err := client.NewClient("", "", kubeconfig)
+	utils.HandleError(err)
+
+	cmd.client = cli.GetSkupperClient().SkupperV1alpha1()
+	cmd.namespace = cli.Namespace
+	cmd.KubeClient = cli.KubeClient
+}
+
+func (cmd *CmdListenerCreate) AddFlags() {
+	cmd.CobraCmd.Flags().StringVarP(&cmd.flags.routingKey, "routing-key", "r", "", "The identifier used to route traffic from listeners to connectors")
+	cmd.CobraCmd.Flags().StringVar(&cmd.flags.host, "host", "", "The hostname or IP address of the local listener")
+	cmd.CobraCmd.Flags().StringVar(&cmd.flags.tlsSecret, "tls-secret", "", "The name of a Kubernetes secret containing TLS credentials")
+	cmd.CobraCmd.Flags().StringVarP(&cmd.flags.listenerType, "type", "t", "tcp", "The listener type. Choices: [tcp].")
+}
+
+func (cmd *CmdListenerCreate) ValidateInput(args []string) []error {
+	var validationErrors []error
+	resourceStringValidator := validator.NewResourceStringValidator()
+	numberValidator := validator.NewNumberValidator()
+	listenerTypeValidator := validator.NewOptionValidator(listenerTypes)
+
+	// Validate arguments name and port
+	if len(args) < 2 {
+		validationErrors = append(validationErrors, fmt.Errorf("listener name and port must be configured"))
+	} else if len(args) > 2 {
+		validationErrors = append(validationErrors, fmt.Errorf("only two arguments are allowed for this command"))
+	} else if args[0] == "" {
+		validationErrors = append(validationErrors, fmt.Errorf("listener name must not be empty"))
+	} else if args[1] == "" {
+		validationErrors = append(validationErrors, fmt.Errorf("listener port must not be empty"))
+	} else {
+		cmd.name = args[0]
+		ok, err := resourceStringValidator.Evaluate(cmd.name)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("listener name is not valid: %s", err))
+		}
+
+		cmd.port, err = strconv.Atoi(args[1])
+		if err != nil {
+			validationErrors = append(validationErrors, fmt.Errorf("listener port is not valid: %s", err))
+		}
```

Comment:
there is no need to do this conversion here, because the `numberValidator` is already doing it

Reply:
I might be missing interpreting your comment above.  NewNumberValidator looks like it needs to be passed an integer.  I don't see how the code is doing the ascii conversion of value being passed to an integer.   I also need the value converted to save in cmd.port to be used in the Run function.  

I removed the above code and passed args[1] directly to the validator
	ok, err = numberValidator.Evaluate(args[1])
		if !ok {
			validationErrors = append(validationErrors, fmt.Errorf("listener port is not valid: %s", err))
		}

if I directly pass arg[1] to the function I get the following errors in unit test
=== RUN   TestCmdListenerCreate_ValidateInput/listener_type_is_not_valid

    listener_create_test.go:267: assertion failed: 

        --- actualErrorsMessages

        +++ test.expectedErrors

          []string{

        - 	"listener port is not valid: value is not an integer",

          	"listener type is not valid: value not-valid not allowed. It shou"...,

          }

Reply:
Oh, sorry for my confusion, you are right. Please ignore this comment.